src/main/java/org/opensearch/ad/util/ExceptionUtil.java

```diff
@@ -116,4 +132,83 @@ public static String getErrorMessage(Exception e) {
             return ExceptionUtils.getFullStackTrace(e);
         }
     }
+
+    /**
+     *
+     * @param exception Exception
+     * @return whether the cause indicates the cluster is overloaded
+     */
+    public static boolean isOverloaded(Throwable exception) {
+        Throwable cause = Throwables.getRootCause(exception);
+        // LimitExceededException may indicate circuit breaker exception
+        return cause instanceof RejectedExecutionException
+            || TransportActions.isShardNotAvailableException(cause)
+            || cause instanceof LimitExceededException;
+    }
+
+    public static boolean isRetryAble(Exception e) {
+        Throwable cause = ExceptionsHelper.unwrapCause(e);
+        RestStatus status = ExceptionsHelper.status(cause);
+        return RETRYABLE_STATUS.contains(status);
```

Comment:
Can this call 155, just in case 155 changes, you dont have to change 2 spots?

Reply:
changed